autoscheduler/scheduler/views.py

```diff
@@ -18,10 +18,10 @@ def _parse_course_filter(course) -> CourseFilter:
     subject = course["subject"]
     course_num = course["courseNum"]
 
-    sections = course["sections"]
+    sections = course.get("sections", [])
 
-    honors = BasicFilter(course["honors"])
-    web = BasicFilter(course["web"])
+    honors = BasicFilter(course.get("honors"))

```

Comment:
Is `.get()` more reliable, or are you just using it to provide a default value?

Reply:
So the `[key]` accessing will throw an error if the key doesn't exist in the data, whereas `.get()` returns `None` if it doesn't exist. This way if `honors`/`web` isn't in the sent data it will still return stuff. This case shouldn't ever actually occur, but it's probably a better practice to use `.get()`